pkg/localstore/subscription_pull.go

```diff
@@ -171,9 +167,9 @@ func (db *DB) SubscribePull(ctx context.Context, bin uint8, since, until uint64)
 		defer db.pullTriggersMu.Unlock()
 
 		for i, t := range db.pullTriggers[bin] {
-			if t == in {
+			if t == trigger {
 				db.pullTriggers[bin] = append(db.pullTriggers[bin][:i], db.pullTriggers[bin][i+1:]...)
```

Comment:
The channel can be closed here.

Reply:
that will trigger the above select, it should be okay like this